zrml/prediction-markets/src/migrations.rs

```diff
@@ -337,6 +349,99 @@ impl<T: Config> OnRuntimeUpgrade for CleanUpStorageForResolvedOrClosedMarkets<T>
     }
 }
 
+impl<T: Config> OnRuntimeUpgrade for MigrateMarketIdsPerBlockStorage<T> {
+    fn on_runtime_upgrade() -> Weight
+    where
+        T: Config,
+    {
+        let mut total_weight = T::DbWeight::get().reads(1);
+
+        if StorageVersion::get::<Pallet<T>>()
+            != PREDICTION_MARKETS_REQUIRED_STORAGE_VERSION_FOR_MIGRATE_MARKET_IDS_STORAGE
+            || utility::get_on_chain_storage_version_of_market_commons_pallet()
+                != MARKET_COMMONS_REQUIRED_STORAGE_VERSION
```

Comment:
Does this migration depend on executing the migration of `MarketCommons` first? Doesn't appear so to me.

Reply:
if MarketCommons has updated deadlines then we can use values from market directly instead of having constant values.

Reply:
yes.